core/include/gnuradio-4.0/PmtCollections.hpp

```diff
@@ -0,0 +1,485 @@
+        ++_size;
+        return *where;
+    }
+
+    void push_back(const T& v) { std::ignore = emplace_back(v); }
+    void push_back(T&& v) { std::ignore = emplace_back(std::move(v)); }
+
+    void pop_back() noexcept {
+        if (_size == 0UZ) {
+            return;
+        }
+        if constexpr (!std::is_trivially_destructible_v<T>) {
+            std::destroy_at(_data + (_size - 1UZ));
+        }
+        --_size;
+    }
+
+    [[nodiscard]] constexpr bool operator==(const vector& other) const { return _size == other._size && std::equal(begin(), end(), other.begin()); }
+    [[nodiscard]] constexpr auto operator<=>(const vector& other) const { return std::lexicographical_compare_three_way(begin(), end(), other.begin(), other.end()); }
+};
+
+namespace detail {
+template<class T, class It>
+inline constexpr bool iter_yields_nonconst_T_ref = std::same_as<std::remove_cvref_t<std::iter_reference_t<It>>, T> && !std::is_const_v<std::remove_reference_t<std::iter_reference_t<It>>>;
+}
+
+template<class T>
+struct vector<T, true> { // managed vector
+    std::pmr::memory_resource* _resource{std::pmr::get_default_resource()};
+    std::size_t                _size{0UZ};
+    std::size_t                _capacity{0UZ};
+    T*                         _data{nullptr};
+
+    vector() = default;
+
+    explicit constexpr vector(std::pmr::memory_resource* mr) noexcept : _resource{mr} {}
+
+    vector(std::size_t n, std::pmr::memory_resource* mr = std::pmr::get_default_resource()) : _resource{mr}, _size{n}, _capacity{n} {
+        if (n > 0) {
+            _data = static_cast<T*>(_resource->allocate(n * sizeof(T), alignof(T)));
+            std::uninitialized_value_construct_n(_data, n);
+        }
+    }
+
+    constexpr vector(std::size_t n, const T& v, std::pmr::memory_resource* mr = std::pmr::get_default_resource()) : _resource{mr}, _size{n}, _capacity{n} {
+        if (n > 0) {
+            _data = static_cast<T*>(_resource->allocate(n * sizeof(T), alignof(T)));
+            std::uninitialized_fill_n(_data, n, v);
+        }
+    }
+
+    vector(const vector& other)
+    requires(std::is_copy_constructible_v<T>)
+        : _resource{other._resource}, _size{other._size}, _capacity{other._size} {
+        if (_size > 0) {
+            _data = static_cast<T*>(_resource->allocate(_capacity * sizeof(T), alignof(T)));
+            if constexpr (std::is_trivially_copyable_v<T>) {
+                std::memcpy(_data, other._data, _size * sizeof(T));
+            } else {
+                std::uninitialized_copy_n(other._data, _size, _data);
+            }
+        }
+    }
+
+    vector(vector&& other) noexcept : _resource{other._resource}, _size{other._size}, _capacity{other._capacity}, _data{other._data} {
+        other._data     = nullptr;
+        other._size     = 0;
+        other._capacity = 0;
+    }
+
+    vector& operator=(const vector& o)
+    requires(std::is_copy_constructible_v<T>)
+    {
```

Comment:
Not strong exception safe -- if allocate throws an exception, we have lost the old data.